payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardNumberController.kt

```diff
@@ -122,14 +124,20 @@ internal class DefaultCardNumberController(
     override val selectedCardBrandFlow: StateFlow<CardBrand> = combineAsStateFlow(
         mostRecentUserSelectedBrand,
         brandChoices,
-    ) { previous, choices ->
-        when (previous) {
-            CardBrand.Unknown -> previous
-            in choices -> previous ?: CardBrand.Unknown
-            else -> {
-                val firstAvailablePreferred = preferredBrands.firstOrNull { it in choices }
-
-                firstAvailablePreferred ?: CardBrand.Unknown
+    ) { previous, allChoices ->
```

Comment:
Could we extract this logic, and unit test it?

Reply:
Yeah, good idea!